pkg/operator/status.go

```diff
@@ -157,6 +125,18 @@ func (optr *Operator) statusFailing(error string) error {
 	return optr.syncStatus(co, conds)
 }
 
+func newClusterOperatorStatusCondition(conditionType osconfigv1.ClusterStatusConditionType,
+	conditionStatus osconfigv1.ConditionStatus, reason string,
+	message string) osconfigv1.ClusterOperatorStatusCondition {
+	return osconfigv1.ClusterOperatorStatusCondition{
+		Type:               conditionType,
+		Status:             conditionStatus,
+		LastTransitionTime: metav1.Now(),
```

Comment:
Since we're updating the LastTransitionTime on these statuscondition objects all the time now, we need to ensure we don't update the LastTransitionTime when we don't desire.  EG, we don't want to update this on available every time the loop runs.  We should add some unit testing around this.

Reply:
@michaelgugino it will not update when when the condition status is not changing. I already added a unit test for the same.
https://github.com/openshift/machine-api-operator/blob/master/vendor/github.com/openshift/cluster-version-operator/lib/resourcemerge/os.go#L52